### ISSUE TITLE
Playing this game in a web browser?

### ISSUE THREAD

**Issue by willwoodlief** [ekolis/FrEee#54]
**What do you want to see?**
I would like to see this game be compiled to Web Assembly using Blazor; and allow the AI to be optionally additionally augmented by javascript, as well as c# scripts

**How would it help players?**
It would make this game truly platform independent; would make it easier for websites to host this as a multiplayer game, and with a little tweaking would allow modding development in the browser window

**Alternatives**
There may be other ways to compile this to javascript, other than Blazor

**Screenshots and mockups**
None yet

**Things to consider**
Posting this idea to see how feasible it may be.  I don't know if there is any complications from doing it

**Comment by ekolis** [comment]
Hmm, that sounds coo, but I've never used Blazor before. So I guess the game host would have to set up a website to host the game or something, and install any necessary mods/shipsets there? And then everyone would connect to it in a browser?

**Comment by willwoodlief** [comment]
I will look more into Blazor.

 Perhaps the hypothetical game host could have multiplayer and single
player ability.

In the multiplayer,  each account could control/install the mods/shipsets
for their current network game. When the game host ran the turn, it would
load in the modlist and run the game using headless chrome

As for single player mode, a user could run each turn themselves, in the
browser, and load the mods they wanted directly in the webpage,  so the
game site would not have to track/build all of that. But, the host could
keep a list of current mods that the user has on a favorite or installed
list, and alert the players when something has been updated ( New
AAA-Shipset version 1.3 released!) for example

**Comment by thegreatpl** [comment]
A client might be possible. I'm not sure about the server even in single player. There are a lot of code that I think might not be compatitble with Blazor. 

If we were going to build a webclient, I would suggest something more along the lines of Unity than Blazor, as we could replace the winforms and wpf clients at the same time with the same code base.

**Comment by willwoodlief** [comment]
Okay, that is what I was wondering -- if an automatic, extra, build step
could have been made that would have compiled the entire project to js. So
there would have been a js version as well as the regular version.  I did
not know if some code changes had to be made, to make it possible, or if
this would be too many changes.  Sounds like a web client would be a new
sub project !

**Comment by thegreatpl** [comment]
Nope. The current client uses IO pretty heavily. Every turn, it saves to disk (might not in single player). That alone would screw up most web conversions to my knowledge. 

On the subject of AI in javascript, that might be possible using [ClearScript](https://github.com/microsoft/ClearScript), but in that case, the AI would have to be entirely written in javascript (it has to be entirely written in Python or C# currently, and cannot mix).
